internal/lock.go

```diff
@@ -149,11 +160,131 @@ func (l *Lock) Download(dir string, tags []string, notags []string, perm string)
 		return fmt.Errorf("nothing to download")
 	}
 	errorCh := make(chan error, total)
-	for _, r := range filteredResources {
+
+	byteCh := make(chan int64)
+	resourceSizes := make([]int64, 0, len(filteredResources))
+
+	if bar {
+		startTime := time.Now()
+
+		//Loop through resources, fetching their metadata and totalling up their sizes in bytes.
```

Comment:
nit: it's really a detail but please follow the standard golang comment format and add a space after `//`.